fetch/handler.go

```diff
@@ -95,11 +96,27 @@ func (h *handler) handleLayerOpinionsReq(ctx context.Context, req []byte) ([]byt
 		h.logger.WithContext(ctx).With().Warning("failed to get prev agg hash", lid, log.Err(err))
 		return nil, err
 	}
-	lo.Cert, err = layers.GetCert(h.cdb, lid)
+
+	certs, err := certificates.Get(h.cdb, lid)
 	if err != nil && !errors.Is(err, sql.ErrNotFound) {
 		h.logger.WithContext(ctx).With().Warning("failed to get certificate", lid, log.Err(err))
 		return nil, err
 	}
+	if err == nil {
+		numValids := 0
+		for _, cert := range certs {
+			if cert.Valid {
+				numValids++
+				if lo.Cert == nil && cert.Cert != nil {
+					lo.Cert = cert.Cert
+				}
+			}
+		}
+		if numValids > 1 {
+			lo.Cert = nil
+		}
```

Comment:
This for loop can be modified to terminate earlier if it finds more than one valid cert:

```suggestion
		var validCert *types.Certificate
		for _, cert := range certs {
			if !cert.Valid {
				continue
			}
			if validCert != nil {
				validCert = nil
				break
			}
			validCert = cert.Cert
		}
		lo.Cert = validCert
```